### PR TITLE
Create method for PUM #18

### LINKED ISSUE #18: Create method for PUM
Compute the Preliminary Unlocking Matrix from Condition Met Vector and Logical Connector Matrix

### PR DESCRIPTION
**Pull request by paulinegarelli**

### CODE REVIEW

File: decide/src/main/java/org/example/Decide.java [hunk 1]
```diff
@@ -37,8 +37,26 @@ public Boolean[] getConditionMetVector() {
         };
     }
 
+    public Boolean[][] computePreliminaryUnlockingMatrix(Boolean[] conditionMetVector) {
+        Boolean[][] preliminaryUnlockingMatrix = new Boolean[15][15];
+        for (int i = 0; i < 15; i++) {
+            for (int j = 0; j <= i; j++) {
+                boolean b = false;
+                switch (settings.LCM[i][j]) {
```
Comment (hampfh) [inline decide/src/main/java/org/example/Decide.java]: (Not important) A cool thing you can do here is actually assign b to the switch, that would look something like this:
```java
boolean b = switch (settings.LCM[i][j]) {
    case NOT_USED -> true;
    case AND -> conditionMetVector[i] && conditionMetVector[j];
    case OR -> conditionMetVector[i] || conditionMetVector[j];
}
```

File: decide/src/main/java/org/example/Decide.java [hunk 2]
```diff
@@ -37,8 +37,26 @@ public Boolean[] getConditionMetVector() {
         };
     }
 
+    public Boolean[][] computePreliminaryUnlockingMatrix(Boolean[] conditionMetVector) {
+        Boolean[][] preliminaryUnlockingMatrix = new Boolean[15][15];
+        for (int i = 0; i < 15; i++) {
+            for (int j = 0; j <= i; j++) {
```
Comment (hampfh) [inline decide/src/main/java/org/example/Decide.java:45]: Nice optimization!

**Review by hampfh — APPROVED** [review]
Looks great! Elegant solution!

File: decide/src/main/java/org/example/Decide.java (hunk 1 above)
Comment (paulinegarelli) [reply decide/src/main/java/org/example/Decide.java]: Yes good thinking, doing that thanks!

**Review by hampfh — APPROVED** [review]
Approved but you will need another review after fixing the merge conflicts

**Review by hampfh — APPROVED** [review]
👍

**Pull request merged**